go/vt/vtgate/planbuilder/testdata/foreignkey_cases.json

```diff
@@ -1523,7 +1523,7 @@
                   "Sharded": false
                 },
                 "FieldQuery": "select 1 from u_tbl8 left join u_tbl9 on u_tbl9.col9 = cast('foo' as CHAR) where 1 != 1",
-                "Query": "select 1 from u_tbl8 left join u_tbl9 on u_tbl9.col9 = cast('foo' as CHAR) where not (u_tbl8.col8) <=> (cast('foo' as CHAR)) and (u_tbl8.col8) in ::fkc_vals and u_tbl9.col9 is null limit 1 for share nowait",
+                "Query": "select 1 from u_tbl8 left join u_tbl9 on u_tbl9.col9 = cast('foo' as CHAR) where u_tbl9.col9 is null and cast('foo' as CHAR) is not null and not (u_tbl8.col8) <=> (cast('foo' as CHAR)) and (u_tbl8.col8) in ::fkc_vals limit 1 for share nowait",
```

Comment:
`cast('foo' as CHAR) is not null` this should be simplified and removed from the predicate.

Reply:
this is left in the query because it is a unsharded SELECT query, and for those we don't touch the AST. 

We could change the logic and force these queries that are part of the bigger foreign key query to be planned with full analysis and rewriting, if you think that is needed.